src/dbnode/integration/setup.go

```diff
@@ -421,7 +422,8 @@ func NewTestSetup(
 			blockRetrieverMgr := block.NewDatabaseBlockRetrieverManager(
 				func(md namespace.Metadata, shardSet sharding.ShardSet) (block.DatabaseBlockRetriever, error) {
 					retrieverOpts := fs.NewBlockRetrieverOptions().
-						SetBlockLeaseManager(blockLeaseManager)
+						SetBlockLeaseManager(blockLeaseManager).
+						SetCacheBlocksOnRetrieve(true)
```

Comment:
Hm, can we avoid setting this to `true` for all tests? 

By default we set this to false for all open source builds now so it's not really indicative of the broadly deployed set of configurations is (and would rather keep the majority of the integration tests representative of that if possible).